astropy/units/quantity.py

```diff
@@ -275,6 +275,14 @@ def __new__(cls, value, unit=None, dtype=None, copy=True, order=None,
     def __array_finalize__(self, obj):
         self._unit = getattr(obj, '_unit', None)
 
+        # Copy info (except for a scalar output for performance reasons) if the
+        # original had `info` defined.  Because of the way the InfoDescriptor
+        # works, `'info' in obj.__dict__` is False until the `info` attribute
+        # is accessed or set.  Note that `obj` can be an ndarray which doesn't
+        # have a `__dict__`.
+        if self.ndim > 0 and hasattr(obj, '__dict__') and 'info' in obj.__dict__:
```

Comment:
Excellent, thanks.

So now for the elephant in the room, do you have any clue what's going on in the case of a mask getitem where it is losing 500 microsecs (not nano sec) when there is an info?  I'll look at this later (probably tonight) but just wondering if you had any immediate insight because I'm stumped.